### ISSUE TITLE
 `sys.modules[__name__]` does not get resolved

### ISSUE THREAD

**Issue by nkakouros** [python/mypy#8683]
Note: if you are reporting a wrong signature of a function or a class in
the standard library, then the typeshed tracker is better suited
for this report: https://github.com/python/typeshed/issues

Please provide more information to help us understand the issue:

* Are you reporting a bug, or opening a feature request?
Feature request? I think this might be out of scope of mypy, but I thought I reported it anyway.

* Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.

In a module file:

```python
variable = 1
current_module = sys.modules[__name__]
current_module.variable
```

* What is the actual behavior/output?

```
(mypy) Module has no attribute "variable"
```

* What is the behavior/output you expect?
I guess, there should be no error?

* What are the versions of mypy and Python you are using?
  Do you see the same issue after installing mypy from Git master?
MyPy:
`mypy 0.770+dev.b8bdc26127d7b7f3ac42b308350b0ec0fba01fcc` (master branch)

Python:
`Python 3.8.1`

**Comment by JelleZijlstra** [comment]
This sort of code is generally too dynamic to support. We could make it pass by changing the type of `sys.modules` in the typeshed project to be something like `Dict[str, Any]` instead of `Dict[str, ModuleType]`, but I'm not sure that's a positive change.

**Comment by ilevkivskyi** [comment]
It is unlikely this will ever be supported.
